packages/block-editor/src/utils/object.js

```diff
@@ -111,3 +111,24 @@ export function setImmutably( object, path, value ) {
 
 	return newObject;
 }
+
+/**
+ * Helper util to return a value from a certain path of the object.
+ * Path is specified as either:
+ * - a string of properties, separated by dots, for example: "x.y".
+ * - an array of properties, for example `[ 'x', 'y' ]`.
+ * You can also specify a default value in case the result is nullish.
+ *
+ * @param {Object}       object       Input object.
+ * @param {string|Array} path         Path to the object property.
+ * @param {*}            defaultValue Default value if the value at the specified path is nullish.
+ * @return {*} Value of the object property at the specified path.
+ */
+export const getValueFromObjectPath = ( object, path, defaultValue ) => {
```

Comment:
Not for this PR, but I saw the other implementations of this in other packages and some of them are a bit different. For example no `Array.isArray( path )` check.. Shouldn't they be identical?

Reply:
That's a valid observation, @ntsekouras 👍 

As I mentioned in the PR description, I'm planning to spend an iteration to unify them all to use the same function, and I'll make sure that the unified version will support all use cases and will be thoroughly covered by unit tests. It's worth noting that there are a few other artifacts throughout the codebase that could use a similar unification. They're all on my radar and I'll be addressing them in an incremental manner. It's a compromise we made during the Lodash migration to not excessively block it, and I'm planning to make sure to clean up the consequences 😉